### ISSUE TITLE
Create "On this page" component

### ISSUE THREAD

**Issue by esizer** [GCTC-NTGC/gc-digital-talent#2818]
## Summary

We currently have a component that allows users to jump to specific sections on the user profile pages. We will need this abstracted into its own component to be used in other areas of the project. This should be placed in `common`.

https://github.com/GCTC-NTGC/gc-digital-talent/blob/7aed9437b851419fd06e7b1e236b1fd2e99fc74f/frontend/talentsearch/src/js/components/profile/ProfilePage/ProfilePage.tsx#L181

@yonikid15 mentioned [SkipTo](https://paypal.github.io/skipto/) as something to reference when building this.

## Screenshot

![Screenshot 2022-05-19 114459](https://user-images.githubusercontent.com/4127998/169341761-fb26a11d-a55f-4807-8543-ac557ccd8b8a.jpg)

## Acceptance Criteria

- [ ]  Abstract "On this page" from `talentsearch/src/js/components/profile/ProfilePage/ProfilePage.tsx`
- [ ] Look into #2703 (optional?)
- [ ] Create tests
- [ ] Storybook

## Notes

We currently have a bug in Firefox and Safari where the links jump the user to the top of the page rather than to the appropriate section #2703. It would be nice if we could resolve this as part of the acceptance criteria as well.

**Comment by patcon** [comment]
Note: not an accessibility-related feature. (@esizer mentioned that screenreaders build this tree themselves from page markup)

So we should have screen-readers ignore this?

**Comment by patcon** [comment]
Removed myself as I was just doing a review :)
